ldb/admin.py

```diff
@@ -60,10 +66,126 @@ def __init__(self, *args, **kwds):
+        """
+        try:
+            instance = Person.objects.get(
+                # These should be unique values.
+                netid=str(row['netid']),
+                # Cast to string as it is saved like this in the DB
+                student__student_number=str(row['student__student_number'])
+            )
+        except Person.DoesNotExist:
+            instance = None
+
+        if instance:
+            return instance, False
+        else:
+            return self.init_instance(row), True
+
+    def import_obj(self, obj, data, dry_run):
+        """
+        Traverses every field in this Resource and calls
+        :meth:`~import_export.resources.Resource.import_field`. If
+        ``import_field()`` results in a ``ValueError`` being raised for
+        one of more fields, those errors are captured and reraised as a single,
+        multi-field ValidationError."""
+        errors = {}
+
+        for field in self.get_import_fields():
+            if isinstance(field.widget, widgets.ManyToManyWidget):
+                continue
+            try:
+                self.import_field(field, obj, data)
+            except ValueError as e:
+                errors[field.attribute] = ValidationError(
+                    force_text(e), code="invalid")
+
+        # Student validation
+        obj.student.yearbook_permission = bool(data['student__yearbook_permission'])
+        obj.student.first_year = data['student__first_year']
+        obj.student.study = str(data['student__study'])
+        obj.student.student_number = str(data['student__student_number'])
+        obj.student.emergency_phone = str(data['student__emergency_phone'])
+        obj.student.emergency_name = str(data['student__emergency_name'])
+
+        try:
+            obj.student.full_clean(exclude='person')
+        except ValidationError as e:
+            for key, value in e:
+                if key is not 'person':
+                    errors["student__" + key] = ValidationError(force_text(value), code="invalid")
+
+        # Member validation
+        obj.member.amount_paid = data['member__amount_paid']
+        obj.member.date_from = datetime.datetime.today()
+
+        try:
+            obj.member.full_clean(exclude='person')
+        except ValidationError as e:
+            for key, value in e:
+                if key is not 'person':
+                    errors["member__" + key] = ValidationError(force_text(value), code="invalid")
+
+        if errors:
+            raise ValidationError(errors)
+
+    def save_instance(self, instance, using_transactions=True, dry_run=False):
+        self.before_save_instance(instance, using_transactions, dry_run)
+        if not using_transactions and dry_run:
+            # we don't have transactions and we want to do a dry_run
+            pass
+        else:
+            instance.save()
+            # Saving student
+            instance.student.person_id = instance.id
```

Comment:
Maybe don't override the `save_instance` and do the extra saves in the `after_save_instance`?

Reply:
Does that make more sense? I thought that `after_save_instance` should be more for things not related to any saving to the DB.

Reply:
Advantage of adding it to existing hooks is that you don't have to copy the boilerplate of the method.